Expensify/App
# Introduce "Mark as unread" functionality in Expensify.cash

@tgolen @marcaaron @roryabraham, this is the new PR! 

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/147475

### Tests
1. Log into e.cash and open a conversation with existing messages
2. On the hover menu of one of the messages, select "mark as unread":
<img width="1233" alt="Screen Shot 2021-03-16 at 3 43 38 PM" src="https://user-images.githubusercontent.com/19366280/111328469-8e7ccc00-866e-11eb-8f24-eed1d7d87d5b.png">
3. The `new` marker should show up over that message and the chat in the LHN should become bolded:
<img width="1602" alt="Screen Shot 2021-03-16 at 3 43 54 PM" src="https://user-images.githubusercontent.com/19366280/111328582-a6ece680-866e-11eb-90c7-b94ecb1e76b5.png">

5. Switch chats. It should remain bolded
6. Switch back to the original chat. The `new` marker should be correctly in place and the chat in the LHN should unbold instantly. The `new` marker should remain in place after that.
7. Switch out and back in. the marker should be gone.
8. Do some regression testing to make sure the new marker shows correctly for actual new messages
9. Test marking a message as unread and receiving new messages afterwards. The new marker should remain in the correct message.
10. Test marking a message as unread, leaving the chat, and then receiving new messages afterwards. The new marker should remain in the marked message.


### Tested On
Testing on mobile soon (I pretty much destroyed my repository, so it is being a pain to get it to work in mobile again)!


### Screenshots

#### Desktop
<img width="1248" alt="Screen Shot 2021-04-23 at 11 10 09 AM" src="https://user-images.githubusercontent.com/19366280/115848746-7d209f80-a424-11eb-8c04-ca2e4d125667.png">


#### iOS
<img width="512" alt="Screen Shot 2021-04-23 at 10 33 44 AM" src="https://user-images.githubusercontent.com/19366280/115848777-890c6180-a424-11eb-8312-c22cb980e09c.png">
<img width="549" alt="Screen Shot 2021-04-23 at 10 33 50 AM" src="https://user-images.githubusercontent.com/19366280/115848794-8e69ac00-a424-11eb-8785-e34a5ce3f592.png">
<img width="466" alt="Screen Shot 2021-04-23 at 10 33 29 AM" src="https://user-images.githubusercontent.com/19366280/115848824-945f8d00-a424-11eb-8c36-6324eeeacb70.png">


#### Android
<img width="513" alt="Screen Shot 2021-04-23 at 10 50 25 AM" src="https://user-images.githubusercontent.com/19366280/115848842-99244100-a424-11eb-909a-270398e7996a.png">
<img width="560" alt="Screen Shot 2021-04-23 at 10 50 36 AM" src="https://user-images.githubusercontent.com/19366280/115848853-9cb7c800-a424-11eb-99ca-be2a336d3fcc.png">
